thoth/storages/graph/postgres.py

```diff
@@ -3775,6 +3775,98 @@ def create_python_cve_record(
 
                 return False
 
+    def update_missing_flag_package_version(
+        self,
+        package_name: str,
+        package_version: str,
+        index_url: str,
+        value: bool,
+    ) -> None:
+        """Update value of is_missing flag for PythonPackageVersion."""
+        with self._session_scope() as session:
+            subq = (session.query(PythonPackageVersion)
+                    .join(PythonPackageIndex)
+                    .filter(PythonPackageVersion.package_name == package_name)
+                    .filter(PythonPackageVersion.package_version == package_version)
+                    .filter(PythonPackageIndex.url == index_url)
+                    .with_entities(PythonPackageVersion.id)
+                    )
+            (
+                session.query(PythonPackageVersion)
+                .filter(PythonPackageVersion.id.in_(subq))
+                .update({"is_missing": value}, synchronize_session='fetch')
+            )
+
+    def get_all_repositories_using_package(self, package_name: str, index_url: str) -> List[str]:
```

Comment:
- `origin` in general encompass more than just repositories.
- `package` is not accepted, in general we refer to `python_package` for `PythonPackageVersionEntity` and `python_package_version` for `PythonPackageVersion`

Please follow query name conventions: https://github.com/thoth-station/storages/tree/master/docs/conventions and if you think a new case should be added please don't hesitate to open a PR :) 

The closest one I see in this case should be `get_adviser_run_origin_all_python_package_version`, `get_<objects>_all_<attribute_filter>()`, but maybe there is a new for a new case as this queries have several joins.

Also in the query convention there is missing the need to add to the doc string of each query an example. See below:
```
        """Retrieve all origins (e.g. repos URLs) in adviser run for a specific Python Package Version.

        Examples:
        >>> from thoth.storages import GraphDatabase
        >>> graph = GraphDatabase()
        >>> graph.get_adviser_run_origin_all_python_package_version(package_name='numpy', index_url='https://pypi.org/simple')
        [HERE SHOULD GO AN EXAMPLE OF A RESULT]
        """
```

Reply:
What else does origin encompass? I thought we were only using it when we trigger advises to indicate what repository it was triggered by.

Reply:
Actually I got confused, it's only git repos :) my bad. In any case to be coherent with conventions and queries in general: `get_adviser_run_origins_all` should retrieve all origins in adviser run, then filter can be applied: package_name, package_version, index_url, distinct also should be there because if we want to count how many for a specific origin we can reuse logic across queries.